frontend/src/store/catalogue/index.ts

```diff
@@ -0,0 +1,45 @@
+export const CATALOGUE_FETCH = 'Catalogue/FETCH';
+export const CATALOGUE_FETCH_SUCCESS = 'Catalogue/FETCH_SUCCES';
+export const CATALOGUE_FETCH_ERROR = 'Catalogue/FETCH_ERROR';
+
+export interface CatalogueProps {
+    catalogue: string;
+};
+
+
+export const catalogueFetch = () => ({
+    type: CATALOGUE_FETCH,
+   
+});
+
+export const catalogueFetchSuccess = ({ catalogue }: CatalogueProps) => ({
+    type: CATALOGUE_FETCH_SUCCESS,
+    payload: {
+        catalogue,
+    }
+});
+
+export const catalogueFetchError = ({ error }: any) => ({
+    type: CATALOGUE_FETCH_ERROR,
+   
+});
+
+export default function reducer(state: any = {}, action: any= {}) {
```

Comment:
I think it should be some initial state with at least bare state shape at the beginning, selectors would throw an error if you would try to access nested props later 😉  And I would really suggest adding types to Redux, I know that it may feel tedious but overall it's really beneficial to spot early-stage bugs. You can check out my PR how I managed to do it